docker-compose.yml

```diff
@@ -429,10 +415,126 @@ services:
         target: /data/rethinkdb_data
     environment:
       TZ: $TZ
+
+  # Aggregates logs and forwards them to Elasticsearch.
+  logstash:
+    image: docker.elastic.co/logstash/logstash-oss:${ELASTIC_VERSION:-7.6}
+    restart: always
+    container_name: logstash
+    hostname: logstash
+    expose:
+      - 12201/udp
+      - 5044
+    ports:
+      - "127.0.0.1:12201:12201/udp"
+    networks:
+      placeos:
+    volumes:
+      - ./config/logstash/config:/config
+      - ./config/logstash/patterns:/opt/logstash/extra_patterns
+    restart: always
+    labels:
+      container_group: logging
+    logging:
+      driver: json-file
+      options:
+        max-size: 99m
+        labels: container_group
+    command: logstash -f /config
+
+  # Run 'docker-compose run --rm validate-logstash-config' to quickly check the logstash config.
+  validate-logstash-config:
+    container_name: validate-logstash
+    image: docker.elastic.co/logstash/logstash-oss:${ELASTIC_VERSION:-7.6}
+    volumes:
+      - ./config/logstash/config:/config
+    command: logstash -t -f /config
+
+  # Sends all container json-file logs to logstash
+  logspout:
+    image: vincit/logspout-gelf
+    hostname: $MONITOR_HOSTNAME
+    container_name: "logspout"
+    networks:
+      placeos:
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
+    command: gelf://$LOGSTASH_HOST:$LOGSTASH_PORT
+    restart: unless-stopped
+    logging:
+      driver: json-file
+      options:
+        max-size: 99m
+
+  # Gets metrics from host machine and send to elastic
+  metricbeat:
+    image: docker.elastic.co/beats/metricbeat-oss:${ELASTIC_VERSION:-7.6}
+    hostname: $MONITOR_HOSTNAME
+    container_name: "metricbeat"
+    user: root
+    networks:
+      placeos:
+    volumes:
+      - /proc:/hostfs/proc:ro
+      - /sys/fs/cgroup:/hostfs/sys/fs/cgroup:ro
+      - /:/hostfs:ro
+      - /var/run/docker.sock:/var/run/docker.sock:ro
+      - ./config/metricbeat.yml:/usr/share/metricbeat/metricbeat.yml
+    cap_add:
+      - SYS_PTRACE
+      - DAC_READ_SEARCH
+        #command: ["metricbeat", "-e", "--strict.perms=false", "-system.hostfs=/hostfs"]
+    command: ["metricbeat", "-e", "--strict.perms=false", "-system.hostfs=/hostfs", "-E", "output.elasticsearch.hosts=[$ELASTIC_HOST:$ELASTIC_PORT]"]
+    restart: unless-stopped
+    tty: true
+    logging:
+      driver: json-file
+      options:
+        max-size: 99m
+
+  kibana:
+    image: docker.elastic.co/kibana/kibana-oss:${ELASTIC_VERSION:-7.6}
+    restart: always
+    container_name: kibana
+    hostname: kibana
+    expose:
+      - 5601
+    networks:
+      placeos:
+    labels:
+      container_group: logging
+    logging:
+      driver: json-file
+      options:
+        max-size: 99m
+        labels: container_group
+    environment:
+      - NODE_OPTIONS=--max-old-space-size=200 # fixes memory leak (https://github.com/elastic/kibana/issues/5170)
+      #- VIRTUAL_HOST=kibana.${DOMAIN}
```

Comment:
Any note on this?

Reply:
Leftover from using a subdoman, can be removed